### ISSUE TITLE
[Turing Data Story] Desert Island Discs - UK notables and their tunes

### ISSUE THREAD

**Issue by billfinnegan** [alan-turing-institute/TuringDataStories#124]
# Story description 

**Please provide a high level description of the Turing Data Story**
Since 1942, notable people have gone on the [BBC Radio 4 programme Desert Island Discs](https://www.bbc.co.uk/programmes/b006qnmr) to share their eight favourite songs, a book, and a luxury item that they would want if stuck on a desert island. As a lighter data story, it could be fun to explore the themes and trends combining data from open data sets for people and music, and perhaps even predicting what a future celebrity might pick.

**Which datasets will you be using in this Turing Data Story?**
The [BBC website](https://www.bbc.co.uk/programmes/b006qnmr/episodes/player) includes ~2300 episodes - it looks like somebody took a stab at compiling that into a [dataset](https://hcommons.org/deposits/item/hc:20545/) a few years back. There could be different places to get info about songs (year, artist, genre) and people (birth date, profession) but [MusicBrainz](https://musicbrainz.org/) and [Pantheon](https://pantheon.world/) look promising.

**Additional context**
Again, I'm more of an idea person that a data scientist, and this is a long way from my research, but it is something I've been hoping someone would put together (and I think the programme has a big following that might be interested in this kind of thing).

# Ethical guideline 

Ideally a Turing Data Story has these properties and follows the [5 safes framework](http://blog.ukdataservice.ac.uk/access-to-sensitive-data-for-research-the-5-safes/).

- [ ] The analysis you produce is openly available and reproducible.
- [ ] Any data used are [open](https://opendatahandbook.org/guide/en/what-is-open-data/) and have an explicit licence, provenance and attribution.
- [ ] Any data used are not [personal data](https://ico.org.uk/for-organisations/guide-to-data-protection/guide-to-the-general-data-protection-regulation-gdpr/key-definitions/what-is-personal-data/) (i.e. the data is anonymous or anonymised).
- [ ] Any linkage of datasets in your data story does not lead to an increased risk of the personal identification of individuals.
- [ ] The Story must be truthful and clear about any limitations of analysis (and potential biases in data).
- [ ] The Story will not lead to negative social outcomes, such as (but not limited to) increasing discrimination or injustice.

# Current status

 - [x] Scoping of the story
 - [x] Write story outline
 - [ ] Add material to the story notebook
 - [ ] Combine materials into a readable story
 - [ ] Proofread
 - [ ] Request reviews
 - [ ] Address reviews
 - [ ] Merge to master branch.

**Comment by samvanstroud** [comment]
Thanks a lot for this story suggestion @billfinnegan! I really like the opportunities for exploration that this would offer. Firstly some basic analysis of music taste as a function of time, profession, etc, finding out which "notables" have the most similar taste, would be fun.

Additionally, quoting from the creator of the DID dataset you link:
> It should be said that I see this as a dataset about the BBC program, and the choices its creators have made. It is certainly a better representation of the BBC and its choices, more than it is a complete representation of any of the guests themselves.

This also opens up a new angle: analysing the show itself, the representation of different groups on the show, etc.

The analysis would present opportunities for the reader to learn about simple statistical techniques, data visualisation, and RESTful APIs. Of course we could take things further if we wanted to analyse information contained in song lyrics/or spectrograms as suggested by @DavidBeavan.

**Comment by samvanstroud** [comment]
Looking again at the linked dataset, it seems this only contains a decades worth of the show (around 2000-2010), around 400 episodes. It might be interesting to write a simple web scraper as part of the project and extract info for all 2300 episodes on the BBC [website](https://www.bbc.co.uk/programmes/b006qnmr/episodes/player).

**Comment by crangelsmith** [comment]
Notes from the Tech Talk scoping session:

 - Compare with national datasets of music taste (spotify)
 - How events change people choices in time
 - **Spotify has an API** that allow you to access metadata on the the music. ([API documentation link](https://developer.spotify.com/documentation/web-api/))
     - Do genres picked over time change
     - Python specific package: [spotipy](https://spotipy.readthedocs.io/en/2.9.0/)
 - Get in contact the BBC?
 - Do MPs have more embarrassing music tastes than average guests?
    - Do they intentionally choose British artists?
 - How evolution of technology affects(?) people choices.
 - What are the most popular choices: books, songs, bands
 - Could we find a larger dataset from the BBC?
 - BBC ggplot2 style (they have a python equivalent?)
     - Yes, [`altair` <- link](https://altair-viz.github.io).
 - There is former TRF (Anna FitzMaurice) who now works for the data team at BBC and who could perhaps help us get hold of further data/ liaising us with other people

People interested in taking this story forward: Camila, Mishka, Ed, Arielle, Glenn

**Comment by edwardchalstrey1** [comment]
What I meant on the BBC ggplot2 was that they have their own version `bbplot`: https://bbc.github.io/rcookbook/#how_to_create_bbc_style_graphics <- since we're visualising BBC radio 4 data!

**Comment by samvanstroud** [comment]
Hackmd for the meeting this afternoon: https://hackmd.io/HVrToWexR3yqtiF34LCo7w

**Comment by edwardchalstrey1** [comment]
for the classical pieces, could use: https://openopus.org/

**Comment by crangelsmith** [comment]
@all-contributors please add @billfinnegan for ideas

**Comment by billfinnegan** [comment]
Data clean up question: from a scan of the data, some std_name values look a little funny:
- antonio
- vivienne
- rankin
- ambrose
- belita

**Comment by billfinnegan** [comment]
Data clean up: from scan of data, some unusual episodes (multiple guests, retrospective, etc.):
- 3225 | 26/01/1992 | The Desert Island Discs Story
- 3224 | 31/12/2017 | 75 Years of Desert Island Discs
- 3210 | 03/01/2014 | Ant and Dec
- 3194 | 05/02/1952 | Peter Brough & Archie Andrews
- 3169 | 12/08/1952 | Michael Denison & Dulcie Gray
- 3006 | 14/12/1956 | Eight inhabitants from the Ascension Islands in the South Atlantic
- 2900 | 05/01/1959 | Ronnie Boyer & Jeanne Ravel
- 2859 | 02/04/1962 | Frank Launder & Sidney Gilliat
- 2845 | 25/12/1972 | Elsie & Doris Waters
- 2841 | 14/02/1978 | Amadeus String Quartet
- 2836 | 27/12/1984 | Beaux Arts Trio
- 2832 | 15/10/1982 | N.Busch, J.Kuchmy, M.Parry, M.Wilson
- 2742 | 16/05/1951 | Denis Compton & W J Edrich
- 2379 | 25/04/1966 | Bob & Alf Pearson
- 2361 | 29/08/1966 | Morecambe & Wise
- 1774 | 31/01/1978 | Raymond Mander & Joe Mitchenson
- 1699 | 24/07/1979 | Dick Clement & Ian La Frenais
- 1544 | 03/09/1982 | Duke & Duchess of Devonshire
- 1495 | 30/12/1983 | Cindy Buxton & Annie Price
- 1381 | 31/10/1986 | Albert & Michel Roux
- 1370 | 06/02/1987 | Peter Fluck & Roger Law
- 247 | 21/02/2014 | Jayne Torvill and Christopher Dean
- 186 | 02/10/2015 | Laurence Marks and Maurice Gran

**Comment by billfinnegan** [comment]
Data clean up: prefix/suffix that may need to be stripped. Note: stripping away lord (or earl, viscount, marquess) when followed by “of [a location]” probably isn’t going to work. In other cases removing the title will result in a single word for name, which might be honorific rather than an actual surname, so also not match.

admiral
and
archbishop
baroness
bishop
canon
captain 
chief
colonel
commander
commissioner
countess
dame
dfc
dr
dso
duchess
earl
frs
general
group
hah
hon
jar
judge
justice
lady
lieutenant
lord 
lt
madame
marchioness
marquess
most
mp
mrs
obe
om
price
princess
professor
qc
ra
rabbi
rev
reverend 
rm
rt 
signalman
sir
sister
the
vc
very
viscount
wing

**Comment by edwardchalstrey1** [comment]
> Data clean up: prefix/suffix that may need to be stripped. Note: stripping away lord (or earl, viscount, marquess) when followed by “of [a location]” probably isn’t going to work. In other cases removing the title will result in a single word for name, which might be honorific rather than an actual surname, so also not match.
> 
> admiral
> and
> archbishop
> baroness
> bishop
> canon
> captain 
> chief
> colonel
> commander
> commissioner
> countess
> dame
> dfc
> dr
> dso
> duchess
> earl
> frs
> general
> group
> hah
> hon
> jar
> judge
> justice
> lady
> lieutenant
> lord 
> lt
> madame
> marchioness
> marquess
> most
> mp
> mrs
> obe
> om
> price
> princess
> professor
> qc
> ra
> rabbi
> rev
> reverend 
> rm
> rt 
> signalman
> sir
> sister
> the
> vc
> very
> viscount
> wing

As an Englishman I'd like to apologise for the length of this list 😆

**Comment by billfinnegan** [comment]
Draft outline at:
https://hackmd.io/@B9ykkfK4TYCyUIbn6sgNdw/Hy3PXmlPu

**Comment by billfinnegan** [comment]
Based on the episode list on Wikipedia, I've created an alternate list of castaways which includes the link to their wikipedia profile - see DID_castaways_wikipedia.csv in Google drive. I also used this as a chance to sort out the multi-guest episodes indicated above (although I left the "Eight inhabitants from the Ascension Islands in the South Atlantic" as I couldn't find any more details about them), so only one person per row with the episode date and link repeated. This goes all the way to the end of May and has a castaway count of 3294, compared to Andrew's episode count of 3225 up to Feb 2020.
